### ISSUE TITLE
Failed upgrade from Nextcloud 19 to 20

### ISSUE THREAD

**Issue by maxidorius** [nextcloud/mail#6057]
### Steps to reproduce

Upgrade from Nextcloud 19.0.13.1 to 20.0.14.1 with the mail app installed

### Expected behavior

Upgrade works sucessfully

### Actual behavior

Upgrade fails and crash the nextcloud upgrade process with:
```
app_1  | Checking for update of app mail in appstore
app_1  | Update app mail from appstore
app_1  | An unhandled exception has been thrown:
app_1  | Error: Class 'OCA\Mail\AppInfo\BootstrapSingleton' not found in /var/www/html/custom_apps/mail/lib/AppInfo/Application.php:35
app_1  | Stack trace:
app_1  | #0 /var/www/html/lib/private/ServerContainer.php(108): OCA\Mail\AppInfo\Application->__construct()
app_1  | #1 /var/www/html/lib/private/ServerContainer.php(171): OC\ServerContainer->getAppContainer('mail', 'Mail')
app_1  | #2 /var/www/html/lib/private/ServerContainer.php(138): OC\ServerContainer->getAppContainerForService('OCA\\Mail\\Migrat...')
app_1  | #3 /var/www/html/lib/private/DB/MigrationService.php(449): OC\ServerContainer->query('OCA\\Mail\\Migrat...')
app_1  | #4 /var/www/html/lib/private/DB/MigrationService.php(473): OC\DB\MigrationService->createInstance('1050Date2020062...')
app_1  | #5 /var/www/html/lib/private/DB/MigrationService.php(419): OC\DB\MigrationService->executeStep('1050Date2020062...', false)
app_1  | #6 /var/www/html/lib/private/legacy/OC_App.php(994): OC\DB\MigrationService->migrate()
app_1  | #7 /var/www/html/lib/private/Installer.php(206): OC_App::updateApp('mail')
app_1  | #8 /var/www/html/lib/private/Updater.php(463): OC\Installer->updateAppstoreApp('mail')
app_1  | #9 /var/www/html/lib/private/Updater.php(269): OC\Updater->upgradeAppStoreApps(Array)
app_1  | #10 /var/www/html/lib/private/Updater.php(132): OC\Updater->doUpgrade('20.0.14.2', '19.0.13.1')
app_1  | #11 /var/www/html/core/Command/Upgrade.php(255): OC\Updater->upgrade()
app_1  | #12 /var/www/html/3rdparty/symfony/console/Command/Command.php(255): OC\Core\Command\Upgrade->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
app_1  | #13 /var/www/html/3rdparty/symfony/console/Application.php(1000): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
app_1  | #14 /var/www/html/3rdparty/symfony/console/Application.php(271): Symfony\Component\Console\Application->doRunCommand(Object(OC\Core\Command\Upgrade), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
app_1  | #15 /var/www/html/3rdparty/symfony/console/Application.php(147): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
app_1  | #16 /var/www/html/lib/private/Console/Application.php(215): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
app_1  | #17 /var/www/html/console.php(100): OC\Console\Application->run()
app_1  | #18 /var/www/html/occ(11): require_once('/var/www/html/c...')
app_1  | #19 {main}
```

### Mail app version

Cannot determine, as nextcloud is stuck in maintenance release

### Mailserver or service

N/A

### Operating system

Official Docker image

**Comment by ChristophWurst** [comment]
> Cannot determine, as nextcloud is stuck in maintenance release

check `/apps/mail/appinfo/info.xml` for `<version>`.

See https://github.com/nextcloud/mail/issues/3727#issuecomment-766787804. You must have upgraded from a very old version of Mail.

**Comment by maxidorius** [comment]
Indeed it was an older version. In the meantime, I have removed the app manually to allow an upgrade, so I cannot give you the version anymore. But I came all the way from Nextcloud 15 and ran the upgrade for each major version (15 to 16, 16 to 17, etc.).
Apps were updated as well, so whatever the version was, it was linked to the upgrade process as well.

The linked issue is indeed the same, thanks for the pointers. I thought I would at least communicate on this issue even tho I just bypassed it altogether.
